R/covariance_plot.R

```diff
@@ -0,0 +1,48 @@
+#' Helpers for Covariance Plot
+#'
+#' @description get the inputted symmetric matrices row and column labels as numeric time points
+#' @param vcov_matrix (`matrix`)\cr name of the input symmetric matrix.
+#' @param string (`string`)\cr string in the column/row names of vcov_matrix that precedes
+#' the time point value. The default value is NULL,
+#' which assumes that the   column/row names of the input matrix
+#' don't have any character
+#' string other than time point value. If string is specified, this value should appear in the column/row names
+#' of the input matrix separated from the time point value by a period.
+#' @return This function returns a list with of two sets of numbers:
+#' the inputted symmetric matrices row labels and column label time points.
+#' @export
+#'
+#' @examples
+#' h_get_timepoint_vars(vcov_matrix = cholest.cov)
+h_get_timepoint_vars <- function(vcov_matrix, string = NULL) {
+  assert_string(string, null.ok = TRUE)
+  resp_name <- paste(string, sep = "")
+  diag <- upper.tri(vcov_matrix, diag = T)
+  vect <- vcov_matrix[diag]
+  rnames <- rownames(vcov_matrix)
+  cnames <- colnames(vcov_matrix)
+  nmat <- outer(rnames, cnames, paste, sep = ":")
+  names(vect) <- nmat[diag]
+  vcov_matrix_row_label <- unlist(strsplit(names(vect), ":"))[seq(1, length(vect) * 2, by = 2)]
+  vcov_matrix_col_label <- unlist(strsplit(names(vect), ":"))[seq(2, length(vect) * 2, by = 2)]
```

Comment:
since the one code part is used twice, save first, something like:
```suggestion
  split_names <- unlist(strsplit(names(vect), ":"))
  vcov_matrix_row_label <- split_names[seq(1, length(vect) * 2, by = 2)]
  vcov_matrix_col_label <- split_names[seq(2, length(vect) * 2, by = 2)]
```